bin/cron/hoc_student_name_cleanup

```diff
@@ -0,0 +1,196 @@
+require 'cdo/db'
+require 'cdo/properties'
+require 'optparse'
+
+CDO.log.info("Script started at #{Time.now}")
+
+HOC_ACTIVITY = PEGASUS_DB[:hoc_activity]
+START_ID_PROPERTY_KEY = "hoc_activity.name_cleanup.start_id"
+
+# Parse options
+options = {
+  retention_days: 90,
+  retain_after_date: nil, # calculated from retention_days
+  start_id: nil,
+  end_id: nil,
+  batch_limit: 100,
+  sleep_msecs: 100,
+  dry_run: false,
+  log_sql: false,
+}
+OptionParser.new do |opts|
+  opts.banner = <<~BANNER
+    Usage: #{File.basename(__FILE__)} [options]
+
+    This script deletes names stored in the pegasus.hoc_activity table after the
+    configured retention period.
+
+    Options:
+  BANNER
+
+  opts.on('--retention=90',
+    Integer,
+    'Retention period in days. Defaults to 90 days if not specified.'
+  ) do |retention_days|
+    options[:retention_days] = retention_days
+  end
+
+  opts.on('--end-id=1000000',
+    Integer,
+    'Id of last entry to clean up (exclusive). If both this and rentention_days ' \
+    'are specified, this value takes precedence.'
+  ) do |end_id|
+    options[:end_id] = end_id
+  end
+
+  opts.on('--start-id=1',
+    Integer,
+    'Id of first entry to clean up (inclusive). If not specified, the script will ' \
+    'determine where to start based on where the last run left off.'
+  ) do |start_id|
+    options[:start_id] = start_id
+  end
+
+  opts.on('--batch-limit=100',
+    Integer,
+    'Maximum number of rows to return in each batch. Defaults to 100 if not specified.'
+  ) do |batch_limit|
+    options[:batch_limit] = batch_limit
+  end
+
+  opts.on('--sleep=100',
+    Integer,
+    'Amount of time to sleep in milliseconds to sleep between batches. Defaults to ' \
+    '100 msecs if not specified.'
+  ) do |sleep_msecs|
+    options[:sleep_msecs] = sleep_msecs
+  end
+
+  opts.on('--dry-run',
+    'Enables read-only mode where no changes are written to the database'
+  ) do |dry_run|
+    options[:dry_run] = dry_run
+  end
+
+  opts.on('--log-sql',
+    'Logs all sql statements sent to the database'
+  ) do |log_sql|
+    options[:log_sql] = log_sql
+  end
+
+  opts.on('-h', '--help', 'Prints this help message') do
+    puts opts
+    exit
+  end
+end.parse!
+CDO.log.info "Initial parsed options: #{options}"
+
+PEGASUS_DB.loggers << CDO.log if options[:log_sql]
+
+# Get the last row in the db for later use. If there is no last row (i.e. there
+# are no rows in the table), we can exit right away.
+last_row = HOC_ACTIVITY.order(:id).last
+if last_row.nil?
+  CDO.log.info "pegasus.hoc_activity table has no rows, exiting"
+  return
+end
+
+# Calculate the start of the retention window. Rows with a timestamp earlier
+# than this are candidates for cleanup.
+options[:retain_after_date] = Date.today - options[:retention_days]
+
+# If start_id was not specified, read the value from the db
+if options[:start_id].nil?
+  options[:start_id] = Properties.get(START_ID_PROPERTY_KEY) || 0
```

Comment:
In the default options it looks like the default start id is 1 `--start-id=1`, is there a reason the fallback here is 0?

Reply:
Oh wait, I just realized that that was an example and not specifying a default, my mistake

Reply:
This confused Hannah too.  I'm going to change the example for start and id to be 1234567 to make it more clear that it's an example.  Thanks!